src/storage/resourceLoader.ts

```diff
@@ -502,23 +561,20 @@ class LocalResourceLoader extends ResourceLoader {
     // set up to do the schema and topic fetch concurrently.
 
     // A quick lambda to fetch the schemas or return empty [] if no schema registry.
-    const getSchemas = async () => {
-      const schemaRegistry = await this.getSchemaRegistry();
-
-      if (!schemaRegistry) {
-        return [];
-      }
-
-      return await this.getSchemasForRegistry(schemaRegistry);
-    };
 
     // Deep fetch the schemas and the topics concurrently.
-    const [schemas, responseTopics] = await Promise.all([getSchemas(), fetchTopics(cluster)]);
+    const [schemas, responseTopics] = await Promise.all([
+      this.getSchemasForCluster(cluster),
+      fetchTopics(cluster),
+    ]);
 
     return correlateTopicsWithSchemas(cluster, responseTopics as TopicData[], schemas as Schema[]);
   }
 
-  private async getSchemaRegistry(): Promise<LocalSchemaRegistry | undefined> {
+  public async getSchemaRegistryForCluster(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    cluster: LocalKafkaCluster,
+  ): Promise<LocalSchemaRegistry | undefined> {
```

Comment:
Might be able to adjust the abstract base class method to say
```ts
  public abstract getSchemaRegistryForCluster(
    cluster?: KafkaCluster,
  )
```
and then TS won't holler when the `LocalResourceLoader` doesn't implement the `cluster` arg

Reply:
ah, it wasn't hollering at me. It hollers at me when LocalResourceLoader has the argument but then doesn't dereference it. Fixing.